server/lib/chat/chat-service.test.ts

```diff
@@ -719,6 +725,102 @@ describe('chat/chat-service', () => {
     })
   })
 
+  describe('editChannel', () => {
+    const expectItWorks = (updates: EditChannelRequest, result: EditChannelResponse) => {
```

Comment:
would vastly prefer that you put in the object to expect to see in this method, rather than constructing that dynamically. As is, it is hard to verify that this method actually gives the output you expect (you really need a test for this test utility), but if you just plug in the actual final object you expect directly it will be obvious and this method will be far simpler

Reply:
The general test structure should always be:
1) Do a thing
2) Compare the result(s) of that thing to *simple* values to ensure they match

If you do a bunch of dynamic stuff in part 2, it becomes very hard to know whether the tests are working or not. Them passing or failing is not a great indication of whether something is wrong in the code or just wrong in the complex test code you've written, and you greatly increase the risk that your steps to generate a result are just reproducing the code under test in the first place (including all its bugs/mistakes).

We want:

```ts
const result = 2 + 2
expect(result).toEqual(4)
```

**not**

```ts
const result = 2 + 2
expect(result).toEqual(2 + 2)
```